zyte_common_items/components.py

```diff
@@ -285,3 +285,42 @@ class RealEstateArea(Item):
 
     #: Area in the raw format, as it appears on the website.
     raw: str
+
+
+@attrs.define(kw_only=True)
+class Header(Item):
+    """A http header value"""
+
+    #: Name of header
+    name: str
+
+    #: Value of the header
+    value: str
+
+
+@attrs.define(kw_only=True)
+class Navigation(Item):
+    """Describe a HTTP request to load a page"""
+
+    #: Name of the category/page to which the navigation takes
+    name: Optional[str] = None
+
+    #: HTTP URL
+    url: str = attrs.field(converter=url_to_str)
+
+    #: HTTP Method
+    #:
+    #: Values: "GET" "POST" "PUT" "DELETE" "OPTIONS" "TRACE" "PATCH"
```

Comment:
For discussion: Should this be strictly in correspondence with what Zyte API supports? or should it be also be open to other HTTP Methods even when Zyte API doesn't support it (e.g. HEAD)

Reply:
[HEAD is coming to Zyte API soon](https://github.com/zytedata/zde-api-server/pull/341), so it may be worth adding.

Not sure if `CONNECTION` will ever make sense here. Not that I think other methods here make sense for navigation, but I guess including them is safer.

Reply:
I would add all methods defined by the HTTP protocol. Even if Zyte API, Scrapy, or whatever underlying tech don't support it, the items definitions should be  independent from the underlying tech stack.

Reply:
Added both HEAD and CONNECT